### PR TITLE
Fix get_bam and get_primary_bam functions

### PR DESCRIPTION
**Pull request by tristanbrown**
@jkokosar @dblenkus

### CODE REVIEW

File: resdk/query.py [hunk 1]
```diff
@@ -145,9 +145,9 @@ def __getitem__(self, index):
         # pylint: disable=protected-access
         if not isinstance(index, (slice,) + six.integer_types):
             raise TypeError
-        if ((not isinstance(index, slice) and index < 0) or
```
Comment (dblenkus) [inline resdk/query.py]: Don't change the code, if the functionality stays the same.

File: resdk/resolwe.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 import yaml
 # Needed because we mock requests in test_resolwe.py
 from requests.exceptions import ConnectionError  # pylint: disable=redefined-builtin
-from six.moves.urllib.parse import urljoin  # pylint: disable=import-error
```
Comment (dblenkus) [inline resdk/resolwe.py]: Fix the problem instead of disabling the checker.

File: resdk/query.py (hunk 1 above)
Comment (dblenkus) [reply resdk/query.py]: Ok, I see. You have to argue such changes better and put a description in the commit message.

**Comment by tristanbrown** [comment]
Superseded by https://github.com/genialis/resolwe-bio-py/pull/151

**Pull request closed without merging**